system/datatypes/searchPages.php

```diff
@@ -0,0 +1,257 @@
+<?php
+/*
+	Class: searchPage
+	handles hypha search requests
+
+	See Also:
+	<Page>
+*/
+	class searchPage extends Page {
+		function __construct($args) {
+			parent::__construct('', $args);
+			//hypha/search/<searchtext>
+			registerCommandCallback('search', Array($this, 'showsearchresult'));
+        }
+
+		function build() {
+			switch ($this->getArg(0)) {
+				//search/text/<pattern>
+				case 'text':
+          $this->showsearchresult($this->getArg(1)); break;
+				default:
+				  $this->showsearchhelp($this->getArg(1));break;
+			}
+		}
+
+		function showsearchresult($pattern) {
+			// $O_O->isUser determines if the private pages are shown.
+			// only loggedin users can see the private pages as well
+		global $uiLangList;
+		global $isoLangList;
+		global $O_O;
+		$pattern = str_replace('_'," ",$pattern);
+		if ($O_O->isUser()) {
+			$includePrivatePages = true;
+		} else {
+				$includePrivatePages = false;
+			}
+        $context = 100; // determines the length of the context around the pattern in case text is found
+		$responce = hyphaSearchPattern($pattern,$includePrivatePages,$context);
+		//$hyphaHtml->writeToElement('pagename', __('searchresult')); uitgezet bz
+		if ($responce) {
+			$this->html->writeToElement('main', $responce);
+		}
+		else {
+			$this->html->writeToElement('main', __('nosearchresult'));
+		}
+	}
+
+	function showsearchhelp($pattern) {
+    // the command should contain a type and pattern
+    // e.g. /search/test/welcome
+    // if the type is missing this function is called and gives this message
+    // due to the fact that search is always activated via the search button this function is seldom activated
+	global $uiLangList;
+	global $isoLangList;
+	$responce = "Error Internal error:illegal call to search, specify search type";
+	$hyphaHtml->writeToElement('pagename', __('searchresult'));
+	$this->html->writeToElement('main', $responce);
+}
+}
+
+/*
+* Function:
+* hyphaSearchPattern()
+* search pages for pattern
+* Parameters:
+* Pattern - search string
+* private - true   on  = include private pages (user is logged on)
+*           false  off = exclude private pages
+* Return:
+* html formatted list of found file references
+* clickable titel
+* text fragment with search pattern in red
+*
+*/
+
+function hyphaSearchPattern($spattern,$bprivate=false, $icontextlength=100)
```

Comment:
Wouldn't it be better to make this function (and the rest below) methods of the `searchPage` class? That reduces chances of name collision and better indicates the structure of things?